src/blocks/LoginForm/LoginFormBlock.ts

```diff
@@ -0,0 +1,106 @@
+import BlockComponent from '../../utils/BlockComponent';
+import loginFormTemplate from './LoginFormBlock.tmpl';
+import Button from "../../components/Button/Button";
+import Input from "../../components/Input/Input";
+import {
+    validateLogin,
+    validatePass,
+    checkInputData
+} from '../../utils/dataValidators'
+import {LoginFormBlockStateType} from "./LoginFormTypes";
+
+
+export default class LoginFormBlock extends BlockComponent {
+    state: LoginFormBlockStateType;
+    constructor(props: any) {
+        super('div', props);
+        this.state = {
+            login: '',
+            password: '',
+        }
+    }
+
+    isDataValid = () => {
+        Object.entries(this.state).forEach((item: any) => {
+            switch (item[0]) {
+                case 'login':
+                    if(!validateLogin(item[1])) this.children.loginInput.className = ' required';
+                    break;
+                case 'password':
+                    if(!validatePass(item[1])) this.children.loginPassword.className = ' required';
+                    break;
+            }
```

Comment:
Можно написать вот так
```suggestion
        Object.entries(this.state).forEach(([key, value]) => {
            switch (key) {
                case 'login':
                    if(!validateLogin(value)) this.children.loginInput.className = ' required';
                    break;
                case 'password':
                    if(!validatePass(value)) this.children.loginPassword.className = ' required';
                    break;
            }
```
Это читаемее, чем чтение по индексу